Microsoft.DotNet.Interactive/KernelBase.cs

```diff
@@ -60,55 +62,73 @@ private void AddDirectiveMiddlewareAndCommonCommandHandlers()
                 (command, context, next) =>
                     command switch
                         {
-                        SubmitCode submitCode =>
-                        HandleDirectivesAndSubmitCode(
-                            submitCode, 
-                            context,
-                            next),
-
-                        LoadExtension loadExtension =>
-                        HandleLoadExtension(
-                            loadExtension,
-                            context, 
-                            next),
-
-                        DisplayValue displayValue =>
-                        HandleDisplayValue(
-                            displayValue,
-                            context,
-                            next),
-
-                        UpdateDisplayedValue updateDisplayValue =>
-                        HandleUpdateDisplayValue(
-                            updateDisplayValue,
-                            context,
-                            next),
+                            SubmitCode submitCode =>
+                            HandleDirectivesAndSubmitCode(
+                                submitCode,
+                                context,
+                                next),
+
+                            LoadExtension loadExtension =>
+                            HandleLoadExtension(
+                                loadExtension,
+                                context,
+                                next),
+
+                            DisplayValue displayValue =>
+                            HandleDisplayValue(
+                                displayValue,
+                                context,
+                                next),
+
+                            UpdateDisplayedValue updateDisplayValue =>
+                            HandleUpdateDisplayValue(
+                                updateDisplayValue,
+                                context,
+                                next),
+
+                            LoadExtensionFromNuGetPackage loadCSharpExtension =>
+                            HandleLoadExtensionInNuGetPackage(
+                                loadCSharpExtension,
+                                context,
+                                next),
 
                             _ => next(command, context)
                         });
         }
 
+        private async Task HandleLoadExtensionInNuGetPackage(
+            LoadExtensionFromNuGetPackage loadExtensionFromNuGetPackage,
+            KernelInvocationContext invocationContext,
+            KernelPipelineContinuation next)
+        {
+            loadExtensionFromNuGetPackage.Handler = async context =>
+            {
+                if (invocationContext.HandlingKernel is IExtensibleKernel extensibleKernel)
```

Comment:
If the handling kernel is not an extensible kernel, it should be an error, no?

Reply:
Yes added a CommandFailed event if that occurs